ui/js/controllers/jobs.js

```diff
@@ -109,6 +109,48 @@ treeherderApp.controller('ResultSetCtrl', [
             $rootScope.selectedJob = job;
         };
 
+        $rootScope.$on(thEvents.applyNewJobs, function (ev, resultSetId) {
+            if ($scope.resultset.id === resultSetId) {
```

Comment:
This line addition is the difference in this PR, and hopefully reduces the hits on the retrieve endpoint. Is it a viable solution? I do not know yet. This check was borrowed from a similar listener in clonejobs: https://github.com/mozilla/treeherder/blob/d0bfec4c9ea1d3e9cdcadd5872b9c78f2fdacd0e/ui/js/directives/treeherder/clonejobs.js#L902-L903

Reply:
Ok, in my local testing of my old branch vs. this change, it *appears* to reduce the number of endpoint hits of the type Ed describes as `https://treeherder.mozilla.org/api/project/mozilla-inbound/jobs/nnn/` in bug 1435023 from 21 to 2 in my local instance. That being said my local instance is a partial ingestion of jobs on inbound, and it's just me hitting the endpoint.

On a typical inbound with lots of decision tasks, and lots of users, I fear we'd still end up with lots of retrieve endpoint hits. I defer to you guys.